actionview/lib/action_view/renderer/collection_renderer.rb

```diff
@@ -144,7 +144,7 @@ def render_collection(collection, view, path, template, layout, block)
           "render_collection.action_view",
           identifier: identifier,
           layout: layout && layout.virtual_path,
-          count: collection.size
+          count: collection.to_a.size
```

Comment:
Would not this load the collection too early making the preload that happens inside `PreloadCollectionIterator` not happen anymore?

Reply:
No because the `PreloadCollectionIterator` has already been instantiated, and preloading has been [disabled](https://github.com/rails/rails/blob/v6.1.0/actionview/lib/action_view/renderer/collection_renderer.rb#L73) (the relation hasn't been loaded at that point). Then, `to_a` will load the relation without preloading. Finally, the `PreloadCollectionIterator` will [handle the preloading of the associations](https://github.com/rails/rails/blob/v6.1.0/actionview/lib/action_view/renderer/collection_renderer.rb#L83).

Reply:
Great! Thank you for investigating.

Reply:
How about using `collection.length` instead?

```suggestion
          count: collection.length
```

`to_a` for relation will dup extra records.

https://github.com/rails/rails/blob/9c120d7559b964d2a619a75faa9776b8f03c5cf7/activerecord/lib/active_record/relation.rb#L243-L246

Reply:
@kamipo Good suggestion, thanks. The only catch was I had to put a check for the length method because there are tests that rely on passing collections that don't respond to `length`, so I delegate back to `size` if `length` isn't available.